src/orion/benchmark/task/profet/profet_task.py

```diff
@@ -0,0 +1,817 @@
+        size=list(network.parameters())[0].size()[1],
+        h=h.tolist(),
+    )
+
+    tmp_file = checkpoint_file.with_suffix(".tmp")
+    with open(tmp_file, "wb") as file:
+        pickle.dump(state, file, protocol=pickle.DEFAULT_PROTOCOL)
+    tmp_file.rename(checkpoint_file)
+
+
+# BUG in pybnn/util/layers.py, where they have self.log_var be a
+# DoubleTensor while x is a FloatTensor! Here we overwrite the
+# forward method, instead of editing the code directly.
+from pybnn.util.layers import AppendLayer
+
+
+@functools.wraps(AppendLayer.forward)
+def _patched_forward(self, x):
+    log_var = self.log_var.type_as(x)
+    return torch.cat((x, log_var * torch.ones_like(x)), dim=1)
+
+
+AppendLayer.forward = _patched_forward
+
+
+InputType = TypeVar("InputType")
+
+
+class ProfetTask(BaseTask, Generic[InputType]):
+    def __init__(
+        self,
+        benchmark: str,
+        max_trials: int = 100,
+        task_id: int = 0,
+        seed: int = 123,
+        input_dir: Union[Path, str] = None,
+        checkpoint_dir: Union[Path, str] = None,
+        train_config: MetaModelTrainingConfig = None,
+        device: Union[torch.device, str] = None,
+    ):
+        super().__init__(max_trials=max_trials)
+        self.benchmark = benchmark
+        self.task_id = task_id
+        self.seed = seed
+        self.input_dir = Path(input_dir if input_dir else "profet_data")
+        self.checkpoint_dir = Path(checkpoint_dir or self.input_dir / "checkpoints")
+        # The config for the training of the meta-model.
+        # NOTE: the train config is used to determine the hash of the task.
+        # TODO: Also save the train_config in the `configuration` of the task.
+        self.train_config = train_config or MetaModelTrainingConfig()
+
+        # The parameters that have an influence over the training of the meta-model are used to
+        # create the filename where the model will be saved.
+        task_hash_params = dict(
+            benchmark=self.benchmark,
+            task_id=self.task_id,
+            seed=self.seed,
+            **asdict(self.train_config),
+        )
+        logger.info(f"Task hash params: {task_hash_params}")
+        task_hash = compute_identity(**task_hash_params)
+
+        filename = f"{task_hash}.pkl"
+
+        self.checkpoint_file = self.checkpoint_dir / filename
+        logger.info(f"Checkpoint file for this task: {self.checkpoint_file}")
+
+        self.device: torch.device
+        if isinstance(device, torch.device):
+            self.device = device
+        else:
+            self.device = torch.device(
+                device or ("cuda" if torch.cuda.is_available() else "cpu")
+            )
+
+        # TODO: Save those in the configuration dict?
+        self._np_rng: Optional[np.random.RandomState] = None
+        self._torch_rng_state: Optional[Tensor] = None
+        with self.seed_randomness():
+            if os.path.exists(self.checkpoint_file):
+                logger.info(
```

Comment:
Does this part need to be seeded? I think loading the model would be fine without seeding.

Reply:
The contextmanager is wrapping both cases of the if statement, even if only the second has randomness, I'd rather keep it that way.

Reply:
Unless we could pass the seed to get_task_network directly. Is there any other use of global RNG during the execution of get_task_network? Because we are already passing a seed to the function.

Reply:
Surprisingly, loading the model *isnt* fine without seeding: Initializing the weights of the dummy model affects global pytorch rng state, so it also needs to be seeded.

This is fixed now.

Reply:
Oh! So you mean than get_default_architecture [here](https://github.com/EmuKit/emukit/blob/main/emukit/examples/profet/meta_benchmarks/architecture.py) is using the global torch RNG?

Reply:
Yes, in a way: initializing the weights of the layers affects the global torch RNG state.

Reply:
That's quite surprising! Well, let's close this discussion then.